SwissTransferFeatures/NewTransferView/VerifyMailView.swift

```diff
@@ -17,33 +17,53 @@
  */
 
 import InfomaniakCoreSwiftUI
+import InfomaniakDI
+import STCore
 import STResources
 import SwiftUI
+import SwissTransferCore
 import SwissTransferCoreUI
 
-struct VerifyMailView: View {
-    let mail: String
-    let fakeCode = "123456"
+private extension UserFacingError {
+    static let validateMailCodeIncorrect = UserFacingError(errorDescription:
+        STResourcesStrings.Localizable.validateMailCodeIncorrectError)
+}
+
+public struct VerifyMailView: View {
+    @LazyInjectService var injection: SwissTransferInjection
+
+    @Environment(\.openURL) private var openURL
+
+    @EnvironmentObject private var rootTransferViewState: RootTransferViewState
 
-    @State private var codeFieldStyle = SecurityCodeFieldStyle.normal
+    let newUploadSession: NewUploadSession
+
+    @State private var isVerifyingCode = false
+    @State private var error: UserFacingError?
+
+    public init(newUploadSession: NewUploadSession) {
+        self.newUploadSession = newUploadSession
+    }
 
-    var body: some View {
+    public var body: some View {
         VStack(alignment: .leading, spacing: IKPadding.large) {
             Text(STResourcesStrings.Localizable.validateMailTitle)
                 .font(.ST.title)
                 .foregroundStyle(Color.ST.textPrimary)
 
-            Text(STResourcesStrings.Localizable.validateMailDescription(mail))
+            Text(STResourcesStrings.Localizable.validateMailDescription(newUploadSession.authorEmail))
                 .font(.ST.body)
                 .foregroundStyle(Color.ST.textSecondary)
 
-            SecurityCodeTextField(style: $codeFieldStyle) { code in
-                if code == fakeCode {
-                    // Code valide
-                } else {
-                    withAnimation {
-                        codeFieldStyle = .error
-                    }
+            SecurityCodeTextField(error: $error) { code in
+                verifyCode(code)
+            }
```

Comment:
```suggestion
            SecurityCodeTextField(error: $error, completion: verifyCode)
```
You might make it shorter if you want